tests/v2/validation/observability/rbac/stackstate_rbac_test.go

```diff
@@ -0,0 +1,257 @@
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	uiExtensionsRepo = "https://github.com/rancher/ui-plugin-charts"
+	uiGitBranch      = "main"
+	rancherUIPlugins = "rancher-ui-plugins"
+)
+
+const (
+	project                 = "management.cattle.io.project"
+	rancherPartnerCharts    = "rancher-partner-charts"
+	systemProject           = "System"
+	localCluster            = "local"
+	stackStateConfigFileKey = "stackstateConfigs"
+)
+
+type StackStateRBACTestSuite struct {
+	suite.Suite
+	client                        *rancher.Client
+	session                       *session.Session
+	cluster                       *management.Cluster
+	project                       *management.Project
+	stackstateAgentInstallOptions *charts.InstallOptions
+	stackstateConfigs             observability.StackStateConfigs
+}
+
+func (rb *StackStateRBACTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *StackStateRBACTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	rb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in rb")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	cluster, err := clusters.NewClusterMeta(rb.client, clusterName)
+	require.NoError(rb.T(), err)
+	rb.cluster, err = rb.client.Management.Cluster.ByID(cluster.ID)
+	require.NoError(rb.T(), err)
+
+	log.Info("Creating a project and namespace for the chart to be installed in.")
+
+	stackstateProject, err := rb.client.Management.Project.Create(projects.NewProjectConfig(rb.cluster.ID))
+	require.NoError(rb.T(), err)
+
+	require.NoError(rb.T(), err)
+	rb.project = stackstateProject
+
+	_, err = namespaces.CreateNamespace(client, cluster.ID, rb.project.Name, charts.StackstateNamespace, "", map[string]string{}, map[string]string{})
+	require.NoError(rb.T(), err, "Unable to create stackstate namespace.")
+
+	log.Info("Verifying if the ui plugin repository for ui extensions exists.")
+	_, err = rb.client.Catalog.ClusterRepos().Get(context.TODO(), rancherUIPlugins, meta.GetOptions{})
+
+	if k8sErrors.IsNotFound(err) {
+		err = observability.AddExtensionsRepo(rb.client, rancherUIPlugins, uiExtensionsRepo, uiGitBranch)
+		require.NoError(rb.T(), err)
+	} else {
+		require.NoError(rb.T(), err)
+	}
+
+	var stackstateConfigs observability.StackStateConfigs
+	config.LoadConfig(stackStateConfigFileKey, &stackstateConfigs)
+	rb.stackstateConfigs = stackstateConfigs
+
+	log.Info("Crete a node driver with stackstate extensions ui to whitelist stackstate URL")
```

Comment:
I don't think we should have this much logging in the setup of the test suite.

Reply:
I have added this because when there is an error it was difficult to debug. since the setup is extensive, would like to maintain these logs.